### PR TITLE
Rename `MeasurementProcess.samples_computational_basis` to `measures_computational_basis`

### PR DESCRIPTION
**Pull request by AlbertMitjans**
Rename method and add more functionality, such that it returns `True` if the observable is a pauli word containing only Z operations.

### CODE REVIEW

File: pennylane/tape/qscript.py [hunk 1]
```diff
@@ -311,9 +311,9 @@ def measurements(self) -> List[MeasurementProcess]:
         return self._measurements
 
     @property
-    def samples_computational_basis(self):
-        """Determines if any of the measurements are in the computational basis."""
-        return any(o.samples_computational_basis for o in self.measurements)
+    def measures_computational_basis(self):
+        """Determines if any of the measurements measures in the computational basis."""
```
Comment (AlbertMitjans) [inline pennylane/tape/qscript.py:315]:
```suggestion
        """Determines if any of the measurements are in the computational basis."""
```

File: pennylane/_device.py [hunk 2]
```diff
@@ -656,10 +656,12 @@ def default_expand_fn(self, circuit, max_expansion=10):
         """
         # pylint: disable=protected-access
 
-        comp_basis_sampled_multi_measure = (
-            len(circuit.measurements) > 1 and circuit.samples_computational_basis
+        multi_measure_comp_basis_without_obs = len(circuit.measurements) > 1 and any(
+            m.measures_computational_basis and m.obs is None for m in circuit.measurements
```
Comment (AlbertMitjans) [inline pennylane/_device.py:660]: We need to add this `any` statement to keep the same functionality as before (expand the tape only when `obs` is `None`).

**Review by rmoyard — COMMENTED** [review]
It looks great, but I have a couple of questions before approval :+1:

File: pennylane/measurements/measurements.py [hunk 3]
```diff
@@ -364,9 +365,17 @@ def has_decomposition(self):
...
         r"""Bool: Whether or not the MeasurementProcess measures in the computational basis."""
-        return self.obs is None
+        if self.obs is None:
+            return True
+        if isinstance(self.obs, qml.PauliZ):
+            return True
+        if isinstance(self.obs, Tensor):
+            return all(isinstance(o, qml.PauliZ) for o in self.obs.obs)
+        if isinstance(self.obs, Prod):
+            return all(isinstance(o, qml.PauliZ) for o in self.obs)
```
[the remark this answers is not in the document]
Comment (AlbertMitjans) [reply pennylane/measurements/measurements.py:377]: We don't support measurements that contain`Sum` and `SProd` operators

File: pennylane/tape/tape.py [hunk 4]
```diff
@@ -30,8 +30,16 @@ class TapeError(ValueError):
     """An error raised with a quantum tape."""
 
 
-def _err_msg_for_some_meas_not_qwc(measurements):
+def non_qwc_error_message(measurements):
     """Error message for the case when some operators measured on the same wire are not qubit-wise commuting."""
+    if any(isinstance(m, (ProbabilityMP, SampleMP, CountsMP)) for m in measurements):
+        return (
+            "Only observables that are qubit-wise commuting "
```
[the remark this answers is not in the document]
Comment (AlbertMitjans) [reply pennylane/tape/tape.py:37]: This error message was copied over from below. We're basically saying that the observables should be Pauli words that don't commute with each other.

Why do you want to remove "Pauli words"? Do we actually support non-pauli words as measurements?

**Pull request closed without merging**